### ISSUE TITLE
Running MXNet on 2 GPUs is doing twice the work

### ISSUE THREAD

**Issue by arnaudfroidmont** [bytedance/byteps#99]
OS: Red Hat 7.6
Running the docker image provided for MXNET. On 1 GPU, it is running fine. When I scale to 2 GPUs. It seems to be running the same workload on each GPU. Making it no faster than on 1 GPU. Do I need to mention anything special. 

Here is my configuration:

`export NVIDIA_VISIBLE_DEVICES=0,1  # say you have 4 GPUs
export DMLC_WORKER_ID=0 # your worker id
export DMLC_NUM_WORKER=1 # you only have one worker
export DMLC_ROLE=worker # your role is worker` 

And here are a couple of parts of the log. 

`BytePS launching worker
INFO:root:start with arguments Namespace(batch_size=32, benchmark=1, cpu_train=False, data_nthreads=4, data_train=None, data_train_idx='', data_val=None, data_val_idx='', disp_batches=20, dtype='float32', gc_threshold=0.5, gc_type='none', image_shape='3,224,224', initializer='default', kv_store='device', load_epoch=None, loss='', lr=0.1, lr_factor=0.1, lr_step_epochs='30,60', macrobatch_size=0, max_random_aspect_ratio=0.25, max_random_h=36, max_random_l=50, max_random_rotate_angle=10, max_random_s=50, max_random_scale=1, max_random_shear_ratio=0.1, min_random_scale=1, model_prefix=None, mom=0.9, monitor=0, network='resnet', num_classes=1000, num_epochs=1, num_examples=1281167, num_layers=50, optimizer='sgd', pad_size=0, random_crop=1, random_mirror=1, rgb_mean='123.68,116.779,103.939', test_io=0, top_k=0, warmup_epochs=5, warmup_strategy='linear', wd=0.0001)
INFO:root:start with arguments Namespace(batch_size=32, benchmark=1, cpu_train=False, data_nthreads=4, data_train=None, data_train_idx='', data_val=None, data_val_idx='', disp_batches=20, dtype='float32', gc_threshold=0.5, gc_type='none', image_shape='3,224,224', initializer='default', kv_store='device', load_epoch=None, loss='', lr=0.1, lr_factor=0.1, lr_step_epochs='30,60', macrobatch_size=0, max_random_aspect_ratio=0.25, max_random_h=36, max_random_l=50, max_random_rotate_angle=10, max_random_s=50, max_random_scale=1, max_random_shear_ratio=0.1, min_random_scale=1, model_prefix=None, mom=0.9, monitor=0, network='resnet', num_classes=1000, num_epochs=1, num_examples=1281167, num_layers=50, optimizer='sgd', pad_size=0, random_crop=1, random_mirror=1, rgb_mean='123.68,116.779,103.939', test_io=0, top_k=0, warmup_epochs=5, warmup_strategy='linear', wd=0.0001)
INFO:root:Launch BytePS process on GPU-0
{'DMLC_WORKER_ID': '0', 'BYTEPS_LOCAL_SIZE': '2', 'CUDA_VERSION': '9.0.176', 'DMLC_NUM_WORKER': '1', 'USE_CUDA_PATH': '/usr/local/cuda:/usr/local/cudnn/lib64', 'PATH': '/usr/local/cuda/bin:/usr/local/nvidia/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'LD_LIBRARY_PATH': '/usr/local/cudnn/lib64:/usr/local/cuda/lib64:/usr/local/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/nccl/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'BYTEPS_BASE_PATH': '/usr/local', 'TERM': 'xterm', 'LIBRARY_PATH': '/usr/local/cudnn/lib64:/usr/local/cuda/lib64:/usr/local/cuda/lib64/stubs', 'CUDA_PKG_VERSION': '9-0=9.0.176-1', 'SHLVL': '2', 'DMLC_PS_ROOT_URI': '10.0.0.1', 'NCCL_VERSION': 'd7a58cfa5865c4f627a128c3238cc72502649881', 'BYTEPS_LOCAL_RANK': '0', 'MXNET_VERSION': '1.4.1', 'NVIDIA_VISIBLE_DEVICES': '0,1', 'HOME': '/root', 'EVAL_TYPE': 'benchmark', 'BYTEPS_PATH': '/usr/local/byteps', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', '_': '/usr/bin/python', 'NVIDIA_REQUIRE_CUDA': 'cuda>=9.0', 'HOSTNAME': 'instance-20190904-1902', 'BYTEPS_GIT_LINK': 'https://github.com/bytedance/byteps', 'DMLC_ROLE': 'worker', 'PWD': '/root', 'DMLC_NUM_SERVER': '1', 'DMLC_PS_ROOT_PORT': '1234'}=============0
INFO:root:Launch BytePS process on GPU-1
{'DMLC_WORKER_ID': '0', 'BYTEPS_LOCAL_SIZE': '2', 'CUDA_VERSION': '9.0.176', 'DMLC_NUM_WORKER': '1', 'USE_CUDA_PATH': '/usr/local/cuda:/usr/local/cudnn/lib64', 'PATH': '/usr/local/cuda/bin:/usr/local/nvidia/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'LD_LIBRARY_PATH': '/usr/local/cudnn/lib64:/usr/local/cuda/lib64:/usr/local/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/nccl/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64', 'BYTEPS_BASE_PATH': '/usr/local', 'TERM': 'xterm', 'LIBRARY_PATH': '/usr/local/cudnn/lib64:/usr/local/cuda/lib64:/usr/local/cuda/lib64/stubs', 'CUDA_PKG_VERSION': '9-0=9.0.176-1', 'SHLVL': '2', 'DMLC_PS_ROOT_URI': '10.0.0.1', 'NCCL_VERSION': 'd7a58cfa5865c4f627a128c3238cc72502649881', 'BYTEPS_LOCAL_RANK': '1', 'MXNET_VERSION': '1.4.1', 'NVIDIA_VISIBLE_DEVICES': '0,1', 'HOME': '/root', 'EVAL_TYPE': 'benchmark', 'BYTEPS_PATH': '/usr/local/byteps', 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility', '_': '/usr/bin/python', 'NVIDIA_REQUIRE_CUDA': 'cuda>=9.0', 'HOSTNAME': 'instance-20190904-1902', 'BYTEPS_GIT_LINK': 'https://github.com/bytedance/byteps', 'DMLC_ROLE': 'worker', 'PWD': '/root', 'DMLC_NUM_SERVER': '1', 'DMLC_PS_ROOT_PORT': '1234'}=============1`

`INFO:root:Epoch[0] Batch [39980-40000]	Speed: 45.99 samples/sec	accuracy=1.000000
INFO:root:Epoch[0] Batch [39980-40000]	Speed: 45.98 samples/sec	accuracy=1.000000
INFO:root:Epoch[0] Batch [40000-40020]	Speed: 45.81 samples/sec	accuracy=1.000000
INFO:root:Epoch[0] Batch [40000-40020]	Speed: 45.81 samples/sec	accuracy=1.000000
INFO:root:Epoch[0] Train-accuracy=0.515420
INFO:root:Epoch[0] Time cost=15617.338
INFO:root:Epoch[0] Train-accuracy=0.484104
INFO:root:Epoch[0] Time cost=15617.582`

**Comment by ymjiang** [comment]
That is the duplicate output of both GPUs. The `samples/sec` represents the speed of each GPU, so when calculating the total speed, you need to multiply it by # of GPU. For example, in your case, the total speed is about 91.6 sample/sec.
